### DISCUSSION TITLE
Different icon packs for datetimepicker icons (prev+next and input)

### DISCUSSION THREAD

**Discussion by Sogl** [buefy/buefy#3979] [Ideas]
### Description

I use free fontawesome icons in my project.
My datetimepicker code:
```vue
<b-datetimepicker
    placeholder="Дата публикации"
    icon="calendar"
    :datepicker="{ firstDayOfWeek: 1, dayNames: ruShortDays, monthNames: ruMonth}">
</b-datetimepicker>
```

In result I see prev and next  but calendar icon isn't working:
![image](https://user-images.githubusercontent.com/7556290/83499690-6bf63980-a4c6-11ea-824a-edd4cb28e0bc.png)

Console error:
```
Could not find one or more icon(s) {prefix: "fas", iconName: "calendar"} {}
```

Because calendar in the regular style (far):
https://fontawesome.com/icons/calendar?style=regular

Ok, when I add pack to my code:
```vue
<b-datetimepicker
    placeholder="Дата публикации"
    icon="calendar"
    icon-pack="far"
    :datepicker="{ firstDayOfWeek: 1, dayNames: ruShortDays, monthNames: ruMonth}">
</b-datetimepicker>
```

Result becomes without prev/next:
![image](https://user-images.githubusercontent.com/7556290/83500234-2be38680-a4c7-11ea-8e5c-db6322f0704b.png)

Console:
```
Could not find one or more icon(s) {prefix: "far", iconName: "angle-left"} {}
Could not find one or more icon(s) {prefix: "far", iconName: "angle-right"} {}
```

Chevrons are in solid state only:
https://fontawesome.com/icons/chevron-left?style=solid

P.S. Of course, I can use one pack of icons, but I do not like the solid version of the calendar icon.

### Why Buefy need this feature

User can use different icon packs for input icon and for prev/next icons.

**Comment by service-paradis** [comment]
I dont know what would be the better approach for this. Maybe `icon` props could accept both a string (current behavior) or an object. The object could be used to overwrite default for a specific icon. ex:
```
icon-pack="fas"
icon="calendar"
:icon-prev="{ icon: 'angle-left', pack: 'far' }"
```

What do you think? Do you have another suggestion?

**Comment by Sogl** [comment]
I think the option with prev/next icons from different packs is not so frequent. However, it gives more flexibility if somebody need it.
